jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/ExamplesTableUtil.java

```diff
@@ -38,18 +45,50 @@
         return result;
     }
 
-    public static List<Triple<String, String, String>> convertTriple(ExamplesTable table, String column1Header, String column2Header, String column3Header) {
-        List<Triple<String, String, String>> result = new ArrayList<>();
+    public static List<Triple<String, Object, String>> convertTriple(ExamplesTable table, String column1Header, String column2Header, String column3Header) {
+        List<Triple<String, Object, String>> result = new ArrayList<>();
         List<Parameters> rows = table.getRowsAsParameters();
         rows.forEach(item -> {
             String column1 = item.valueAs(column1Header, String.class);
-            String column2 = item.valueAs(column2Header, String.class);
+            Object column2 = null;
+            Type dataType = getValueType(item, item.valueAs(column2Header, String.class));
+            if (dataType != null) {
+                column2 = item.valueAs(column2Header, dataType);
+            }
             String column3 = item.values().containsKey(column3Header) ? item.valueAs(column3Header, String.class) : null;
             result.add(new ImmutableTriple<>(column1, column2, column3));
         });
         return result;
     }
 
+    private static Type getValueType(Parameters item, String data) {
+        if (item.values().containsKey(ExampleTableConstraints.TYPE)) {
```

Comment:
Nitpicking: how about `return String.class` at the end of the method unconditionally to simplify the two if/else a bit to be a bit more readable? Or some simplification like that...